src/snmpd/patch-5.7.3+dfsg/cross-compile-changes.patch

```diff
@@ -0,0 +1,122 @@
+Index: net-snmp-5.7.3+dfsg/debian/rules
+===================================================================
+--- net-snmp-5.7.3+dfsg.orig/debian/rules
++++ net-snmp-5.7.3+dfsg/debian/rules
+@@ -59,11 +59,11 @@ override_dh_auto_configure:
+ 
+ override_dh_auto_build:
+ 	dh_auto_build
+-	cd python; python setup.py build --basedir=$(CURDIR)
++	cd python; CC=$(CROSS_CC) LDSHARED="$(CROSS_CC) -shared" python setup.py build --basedir=$(CURDIR)
+ 
+ override_dh_auto_install:
+ 	dh_auto_install
+-	cd python; python setup.py install --root=$(CURDIR)/debian/tmp \
++	cd python; CC=$(CROSS_CC) LDSHARED="$(CROSS_CC) -shared" python setup.py install --root=$(CURDIR)/debian/tmp \
+ 		--install-layout=deb --basedir=$(CURDIR) ; \
+ 		find $(CURDIR) -name *.pyc -delete
+ 
+Index: net-snmp-5.7.3+dfsg/perl/ASN/Makefile.PL
+===================================================================
+--- net-snmp-5.7.3+dfsg.orig/perl/ASN/Makefile.PL
++++ net-snmp-5.7.3+dfsg/perl/ASN/Makefile.PL
+@@ -72,6 +72,8 @@ sub InitMakeParams {
+     else
+     {
+       $opts = NetSNMPGetOpts();
++$Params{'CC'} = "$ENV{'CROSS_CC'}";
```

Comment:
For these changes, Perl's MakeMaker reads the `PERL_MM_OPT` environment variable and accepts flags from there. Therefore, in `debian/rules` you should be able to do `export PERL_MM_OPT="CC=$(CROSS_CC) LD=$(CROSS_CC)"` (not sure whether quotes are needed or not).